### DISCUSSION TITLE
Fun fact

### DISCUSSION THREAD

**Discussion by sebastienros** [ncalc/ncalc#230] [General]
A little anecdote about the history of this project. When I first developed it we had a customer who was building gaz turbines with lots of sensors on them (huge company). We were creating a tool for engineers to define custom stats and charts based on raw live data. Their existing solution was done in C++ and the expression evaluation too. The project lead was saying that we would never be able to create something as fast in C#. A few days after I had implemented NCalc and benchmarked it against this C++ library they were using, and NCalc was at least 10 times faster (what I remember, even 100 times maybe) while being much simpler and more flexible. The customer spent some trying to understand what was wrong in NCalc until he accepted that fact. He loved C# but still couldn't believe it. We were also happily surprised.

**Comment by gumbarros** [comment]
Very nice to hear, congrats for the original development of the lib, this lib is very important for me!

I was using `DataTable.Compute` before NCalc and in a benchmark from months ago I discovered it its really faster. 
![image](https://github.com/ncalc/ncalc/assets/52143624/220ce6a1-0d35-4f3f-88d1-4e8f6e2c155c)

**Comment by yallie** [comment]
We're using NCalc for I believe more than 10 years in our product.
Many other open source libraries that we used came and went, but NCalc stayed ❤️
Thanks Sébastien!
